src/service/viewer-impl.js

```diff
@@ -860,12 +860,33 @@ export class Viewer {
     if (!this.hasCapability('fragment')) {
       return Promise.resolve('');
     }
-    return this.sendMessageUnreliable_('fragment', undefined, true).then(
+    return this.sendMessageUnreliable_('getFragment', undefined, true).then(
       hash => hash || ''
     );
   }
 
   /**
+   * Update the fragment of the viewer if embedded in a viewer,
+   * otherwise update the page url fragment
+   * @param {string} fragment
+   * @return {!Promise}
+   */
+  updateFragment(fragment) {
+    if (!this.isEmbedded_) {
+      if (this.win.history.replaceState) {
+        const newUrl = this.win.location.href.split('#')[0] + '#' + fragment;
+        this.win.history.replaceState({}, '', newUrl);
+      }
+      return Promise.resolve();
+    }
+    if (!this.hasCapability('fragment')) {
+      return Promise.resolve();
+    }
+    return /** @type {!Promise} */ (this.sendMessageUnreliable_(
```

Comment:
Yeah, this is not implemented on their side yet but they are going to be able to rewrite their url. This is needed because we want to track users sharing the google.com url.
When a user A shares a news link, it's associated with a share-tracking fragment (for example `www.google.com/amp/some-publisher/some-page.html#.54321`). When another user B enter the above url, the fragment part of the url should change to associate with this user B (`www.google.com/amp/some-publisher/some-page.html#.54321`). So when user B opens `www.google.com/amp/some-publisher/some-page.html#.12345`, we want to change the url to `www.google.com/amp/some-publisher/some-page.html#.54321`. That url replacement can only be done from the viewer side.